pages/books.tsx

```diff
@@ -34,6 +34,19 @@ const Catalog = () => {
 
   const { response } = jsonData.data;
 
+  //search query filters based on all fields, with memoization
+  const filteredItems = useMemo(() => {
+    const keys: string[] = Object.keys(response[0]);
+    return response.filter((item) => {
+      return keys.some((key) => {
+        return (
```

Comment:
I didn't check but just make sure that things look okay on initial data loading! Since we're using `filteredItems` now instead of `response` :)

Reply:
Yes, it works on initial load!